Added @private directive

marshalWilliam opened the pull request:
Implemented custom directive @private

marshalWilliam committed:
Added @private directive

Implemented custom directive @private

ccpacillos commented on schemas/directives/private.ts:
```diff
@@ -0,0 +1,42 @@
+          });
+
+          if (userInfo) {
+            if (await checkEmail(userInfo.email)) {
```
I would suggest retrieving the user by `id` instead of email.

ccpacillos commented on schemas/directives/private.ts:
```diff
@@ -0,0 +1,42 @@
+
+          if (userInfo) {
+            if (await checkEmail(userInfo.email)) {
+              context.isAuth = true;
```
You will need to put user information within your context. For example, `context.user`.

marshalWilliam committed:
Update @private directive

- changed jwt.decode to jwt.verify
- check if user/account exist using ID instead of Email Address
- save contents of user/account information to context.user

ccpacillos commented on schemas/directives/private.ts:
```diff
@@ -0,0 +1,39 @@
+              context.isAuth = true;
+            }
+          }
+
```
Throw an error if user is not found.

ccpacillos commented on schemas/directives/private.ts:
```diff
@@ -0,0 +1,39 @@
+          if (userInfo) {
+            context.user = await checkID(userInfo.user_id);
+            if (context.user) {
+              context.isAuth = true;
```
No need to use `context.isAuth` anymore since you now have `context.user`.

marshalWilliam committed:
Update #2 @private directives

- throwing Invalid Token error when user is not found and ID is not found.
